project/Build.scala

```diff
@@ -1088,93 +1089,117 @@ object Build extends sbt.Build {
+      val sharedTestDir = testDirValue / "../../../shared/src/test"
+
+      def listAllScalaFilesIf(testSubDir: String, condition: Boolean): Seq[File] = {
+        if (condition) {
+          ((testDirValue / testSubDir) ** "*.scala").get ++
+            ((sharedTestDir / testSubDir) ** "*.scala").get
+        } else Nil
+      }
 
-            listAllScalaFilesIf("require-jdk7", javaVersion.value >= 7) ++
-            listAllScalaFilesIf("require-jdk8", javaVersion.value >= 8) ++
-            listAllScalaFilesIf("require-sam", hasSAM)
-          },
+      val hasSAM =
+        scalaBinaryVersion != "2.10" && scalacOptions.contains("-Xexperimental")
 
-          /* Generate a scala source file that throws exceptions in
-             various places (while attaching the source line to the
-             exception). When we catch the exception, we can then
-             compare the attached source line and the source line
-             calculated via the source maps.
+      listAllScalaFilesIf("require-jdk7", javaVersion >= 7) ++
+        listAllScalaFilesIf("require-jdk8", javaVersion >= 8) ++
+        listAllScalaFilesIf("require-sam", hasSAM)
+    }
 
-             see test-suite/src/test/resources/SourceMapTestTemplate.scala
-           */
-          sourceGenerators in Test <+= Def.task {
-            val dir = (sourceManaged in Test).value
-            IO.createDirectory(dir)
-
-            val template = IO.read((resourceDirectory in Test).value /
-              "SourceMapTestTemplate.scala")
-
-            def lineNo(cs: CharSequence) =
-              (0 until cs.length).count(i => cs.charAt(i) == '\n') + 1
-
-            var i = 0
-            val pat = "/\\*{2,3}/".r
-            val replaced = pat.replaceAllIn(template, { mat =>
-              val lNo = lineNo(mat.before)
-              val res =
-                if (mat.end - mat.start == 5)
-                  // matching a /***/
-                  s"if (TC.is($i)) { throw new TestException($lNo) } else "
-                else
-                  // matching a /**/
-                  s"; if (TC.is($i)) { throw new TestException($lNo) } ;"
+    CrossProject.withOutScalaJSPlugin(
+      "testSuiteJVM",
+      "testSuite",
+      file("test-suite"),
+      CrossType.Full
+    ).
+    settings(commonSettings: _*).
+    settings(
+      publishArtifact in Compile := false,
+
+      scalacOptions ~= (_.filter(_ != "-deprecation")),
+      // Need reflect for typechecking macros
+      libraryDependencies +=
+        "org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided",
+
+      testOptions += Tests.Argument(TestFrameworks.JUnit, "-v", "-a")
+    ).
+    jvmSettings(
+      name := "Scala.js test suite on JVM",
+      libraryDependencies += "com.novocode" % "junit-interface" % "0.11" % "test",
+      sources in Test ++= listAdditionalSources("jvm",
+          (sourceDirectory in Test).value, scalaBinaryVersion.value,
+          scalacOptions.value, javaVersion.value)
```

Comment:
If you are basically fully re-implementing a cross project, why do you use one? It seems you are not getting anything from the implementation of `CrossProject`.

If you really want to go down the path of adding `withoutScalaJSPlugin` you need to change the other build definitions we have as well (probably in an earlier commit). However, I'd advocate for implementing cross building ourselves in the Scala.js build itself (just like we do for tools).

This also makes sense, since we probably won't ever have JVM specific sources in this project (no point of testing mainline Scala in the Scala.js build).